lte/gateway/deploy/agw_install_ubuntu.sh

```diff
@@ -0,0 +1,159 @@
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+# Setting up env variable, user and project path
+MAGMA_USER="magma"
+AGW_INSTALL_CONFIG="/etc/systemd/system/multi-user.target.wants/agw_installation.service"
+AGW_SCRIPT_PATH="/root/agw_install_ubuntu.sh"
+DEPLOY_PATH="/home/$MAGMA_USER/magma/lte/gateway/deploy"
+SUCCESS_MESSAGE="ok"
+NEED_REBOOT=0
+WHOAMI=$(whoami)
+MAGMA_VERSION="${MAGMA_VERSION:-v1.5}"
+CLOUD_INSTALL="cloud"
+GIT_URL="${GIT_URL:-https://github.com/magma/magma.git}"
+
+echo "Checking if the script has been executed by root user"
+if [ "$WHOAMI" != "root" ]; then
+  echo "You're executing the script as $WHOAMI instead of root.. exiting"
+  exit 1
+fi
+
+wget https://raw.githubusercontent.com/magma/magma/"$MAGMA_VERSION"/lte/gateway/deploy/agw_pre_check_ubuntu.sh
+if [[ -f ./agw_pre_check_ubuntu.sh ]]; then
+  chmod 644 agw_pre_check_ubuntu.sh && bash agw_pre_check_ubuntu.sh
+  while true; do
+      read -p "Do you accept those modifications and want to proceed with magma installation?(y/n)" yn
+      case $yn in
+          [Yy]* ) break;;
+          [Nn]* ) exit;;
+          * ) echo "Please answer yes or no.";;
+      esac
+  done
+else
+  echo "agw_pre_check_ubuntu.sh is not available in your version"
+fi
+
+echo "Checking if Ubuntu is installed"
+if ! grep -q 'Ubuntu' /etc/issue; then
+  echo "Ubuntu is not installed"
+  exit 1
+fi
+
+echo "Making sure $MAGMA_USER user is sudoers"
+if ! grep -q "$MAGMA_USER ALL=(ALL) NOPASSWD:ALL" /etc/sudoers; then
+  apt install -y sudo
+  adduser --disabled-password --gecos "" $MAGMA_USER
+  adduser $MAGMA_USER sudo
+  echo "$MAGMA_USER ALL=(ALL) NOPASSWD:ALL" >> /etc/sudoers
+fi
+
+echo "Need to check if both interfaces are named eth0 and eth1"
+INTERFACES=$(ip -br a)
+if [[ $1 != "$CLOUD_INSTALL" ]] && ( [[ ! $INTERFACES == *'eth0'*  ]] || [[ ! $INTERFACES == *'eth1'* ]] || ! grep -q 'GRUB_CMDLINE_LINUX="net.ifnames=0 biosdevname=0"' /etc/default/grub); then
+  # changing intefaces name
+  sed -i 's/GRUB_CMDLINE_LINUX=""/GRUB_CMDLINE_LINUX="net.ifnames=0 biosdevname=0"/g' /etc/default/grub
+  sed -i 's/enp0s3/eth0/g' /etc/netplan/50-cloud-init.yaml
+  # changing interface name
+  grub-mkconfig -o /boot/grub/grub.cfg
+
+  # interface config
+  apt install ifupdown
+  echo "auto eth0
+  iface eth0 inet dhcp" > /etc/network/interfaces.d/eth0
+  # configuring eth1
+  echo "auto eth1
+  iface eth1 inet static
+  address 10.0.2.1
+  netmask 255.255.255.0" > /etc/network/interfaces.d/eth1
+  # name server config
+  ln -sf /var/run/systemd/resolve/resolv.conf /etc/resolv.conf
+
+  # get rid of netplan
+  systemctl unmask networking
+  systemctl enable networking
+
+  apt-get --assume-yes purge nplan netplan.i
+
+  # Setting REBOOT flag to 1 because we need to reload new interface and network services.
+  NEED_REBOOT=1
```

Comment:
As we don't change the kernel version I don't think we have to reboot anymore. So we can remove the reboot part and the part when we setup the systemd job

Reply:
interface renaming needs reboot.